runtime/ftplugin/d.lua

```diff
@@ -1,3 +1,3 @@
 vim.bo.commentstring = '// %s'
 
-vim.b.undo_ftplugin = 'setl commentstring<'
+vim.b.undo_ftplugin = vim.b.undo_ftplugin .. ' | setl commentstring<'
```

Comment:
This leads to an error (concatenation with `nil`).

Reply:
> This leads to an error (concatenation with `nil`).

Oh yeah I see, those languages don't have vimscript ftplugins at all so undo_ftplugin was never set, makes sense.

Should that be a null check (something like vim.b.undo_ftplugin or '') or can we assume that those filetypes won't be added in upstream vim and remove the concatenation altogether?

Reply:
I have an idea: instead of ` |`, use `\n` to separate commands in a Lua ftplugin (preferably without a preceding space). That'll allow using `(vim.b.undo_ftplugin or '')`, and will not be treated as a part of an `:unmap` command like `|` does.

Reply:
Maybe it can be `(vim.b.undo_ftplugin or '') .. '\n setl commentstring<'` here instead. `\n` needs to be used instead of ` |` as using `|` without a command is not no-op.

Reply:
I've updated the offending files, should the rest of the lua ftplugins also get that updated concatenation, or, since they do all have corresponding vimscript plugins that fill the undo_ftplugin value, be left as is?
Very unlikely, but if for some reason upstream vim removes an ftplugin or all undo statements in there, that could potentially create the same nil-concatenation error in the future.

Reply:
Let's leave that out of this PR?